rclcpp/include/rclcpp/message_memory_strategy.hpp

```diff
@@ -62,15 +80,52 @@ class MessageMemoryStrategy
     return std::allocate_shared<MessageT, MessageAlloc>(*message_allocator_.get());
   }
 
+  virtual std::shared_ptr<rcl_message_raw_t> borrow_raw_message(unsigned int capacity)
```

Comment:
`size_t` (obviously depends on changing that down the API chain too, but I think it should be)

Reply:
I chose `unsigned int` because RTI only serializes this. 
https://community.rti.com/static/documentation/connext-dds/5.2.3/doc/api/connext_dds/api_cpp/structDDS__DynamicData.html#ae31195afe3fe6f567ddce38e7f99980e

I can change to `size_t` and cast it to unsigned for connext, but this might lead to data loss.

Reply:
`size_t` is unsigned, and usually larger than `unsigned int`. Do you mean the user could ask for something larger and that would be an issue? You could check that it isn't too big at the point where you pass the `size_t` to RTI's API.